src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -49,6 +59,106 @@ public async Task PutJavaScript()
 
             await PutInternal(validatedAsAdmin: false);
         }
+        
+        [RavenAction("/databases/*/admin/indexes/test", "POST", AuthorizationStatus.DatabaseAdmin, DisableOnCpuCreditsExhaustion = true)]
+        public async Task PutTestIndex()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var input = await context.ReadForMemoryAsync(RequestBodyStream(), "Input");
+
+                if (input.TryGet("IndexDefinition", out BlittableJsonReaderObject indexDefinitionBlittable) == false)
+                    throw new Exception();
+                
+                var indexDefinition = JsonDeserializationServer.IndexDefinition(indexDefinitionBlittable);
+
+                if (indexDefinition.Name is null)
+                    indexDefinition.Name = Guid.NewGuid().ToString().Replace("-", string.Empty);
+
+                Index index;
+                
+                switch (indexDefinition.SourceType)
+                {
+                    case IndexSourceType.Documents:
+                        switch (indexDefinition.Type)
+                        {
+                            case IndexType.Map:
+                            case IndexType.JavaScriptMap:
+                                index = MapIndex.CreateNewForTest(indexDefinition, context);
```

Comment:
We have multiple `CreateNewForTest` calls here. I checked them and the only thing we do there additionally is `index.InitializeTestIndex(context)`. Couldn't we call it here after the creation of index instance?

Reply:
Additionally we could move this piece of code which create a test index instance to IndexStore. We already have a very similar code in `HandleStaticIndexChange`. Maybe we could extract some common parts and avoid code duplication.

Reply:
Now we initialize test index inside IndexStore.CreateTestIndexFromDefinition, I also moved duplicated code to IndexStore.CreateIndexFromDefinition